python/beeai_framework/adapters/ollama/backend/chat.py

```diff
@@ -28,6 +28,9 @@ def provider_id(self) -> ProviderName:
         return "ollama"
 
     def __init__(self, model_id: str | None = None, settings: dict | None = None) -> None:
+        if settings is not None and not hasattr(settings, "base_url") and "OLLAMA_BASE_URL" in os.environ:
+            settings["base_url"] = os.getenv("OLLAMA_BASE_URL")
```

Comment:
Instead of mutating the user-passed object, we should create a copy and modified it

Reply:
Do you mean as I have added in [429c4ba](https://github.com/i-am-bee/beeai-framework/pull/389/commits/429c4bab96b50acd94bed6ae0274878638b89dd9)?

Reply:
Ohh sorry, but  does`_settings = settings.copy()` works for `None` value ?